diff --git a/src/brain/QmixBrain.py b/src/brain/QmixBrain.py
index 079a05d..d4b1d04 100644
--- a/src/brain/QmixBrain.py
+++ b/src/brain/QmixBrain.py
@@ -184,4 +184,9 @@ class QmixBrain(BrainBase):
             self.update_target_network(self.fit_conf['tau'], self.qnet, self.qnet2)
             self.update_target_network(self.fit_conf['tau'], self.mixer, self.mixer2)
 
+        # decay epsilon
+        self.eps = self.eps * self.eps_gamma
+        if self.eps <= self.eps_min:
+            self.eps.fill_(self.eps_min.data[0])
+
         return fit_dict
