diff --git a/src/brain/QmixBrain.py b/src/brain/QmixBrain.py
index 079a05d..b93afe5 100644
--- a/src/brain/QmixBrain.py
+++ b/src/brain/QmixBrain.py
@@ -13,7 +13,7 @@ class QmixBrainConfig(ConfigBase):
         self.brain = {
             'optimizer': 'lookahead',
             'lr': 1e-5,
-            'gamma': 1.0,
+            'gamma': 0.9,
             'eps': 0.5,
             'eps_gamma': 0.995,
             'eps_min': 0.01,
@@ -27,7 +27,7 @@ class QmixBrainConfig(ConfigBase):
             'tau': 0.1,
             'auto_norm_clip': True,
             'auto_norm_clip_base_val': 0.1,
-            'norm_clip_val': 1.0
+            'norm_clip_val': None
         }
 
 
@@ -179,9 +179,15 @@ class QmixBrain(BrainBase):
                                    clip_val=norm_clip_val)
 
             fit_dict['loss2'] = dn(loss2)
+            fit_dict['loss_tot'] = dn(loss + loss2)
 
         else:
             self.update_target_network(self.fit_conf['tau'], self.qnet, self.qnet2)
             self.update_target_network(self.fit_conf['tau'], self.mixer, self.mixer2)
 
+        # decay epsilon
+        self.eps = self.eps * self.eps_gamma
+        if self.eps <= self.eps_min:
+            self.eps.fill_(self.eps_min.data[0])
+
         return fit_dict
diff --git a/src/test/test.py b/src/test/test.py
index 4069bdc..1a00b99 100644
--- a/src/test/test.py
+++ b/src/test/test.py
@@ -10,6 +10,7 @@ if __name__ == '__main__':
 
     conf = QmixAgentConfig()
     use_noisy_q = conf.brain.brain['use_noisy_q']
+    gamma = conf.brain.brain['gamma']
 
     agent = QmixAgent(conf)
     if use_noisy_q:
@@ -17,6 +18,7 @@ if __name__ == '__main__':
 
     runner_conf = RunnerConfig(agent=agent)
     runner_conf.runner['num_runners'] = 1
+    runner_conf.runner['gamma'] = gamma
     runner_conf.runner['n_hist_steps'] = conf.fit['hist_num_time_steps']
     runner = RunnerManager(runner_conf)
 
